repos/spack_repo/builtin/packages/py_mdanalysis/package.py

```diff
@@ -89,3 +98,7 @@ class PyMdanalysis(PythonPackage):
     depends_on("py-gsd@1.9.3:", when="@:2.5.0", type=("build", "run"))
     depends_on("py-biopython@1.80:", when="@:2.6.1", type=("build", "run"))
     depends_on("py-networkx@2.0:", when="@:2.6.1", type=("build", "run"))
+
+    def url_for_version(self, version):
+        if version >= Version("2.8.0"):
+            return f"https://files.pythonhosted.org/packages/source/m/mdanalysis/mdanalysis-{version}.tar.gz"
```

Comment:
Why was this necessary?

Reply:
Before 2.8.0 the name of the archive is `MDAnalysis-<version>.tar.gz`

Reply:
Ah! I did not realize this. Thanks for pointing it out.

```suggestion
    def url_for_version(self, version):
        """
        From version 2.8.0 onwards, the archive named changed
        to 'mdanalysis-{version}.tar.gz from 'MDAnalysis-{version}.tar.gz
        """
        if version >= Version("2.8.0"):
            return f"https://files.pythonhosted.org/packages/source/m/mdanalysis/mdanalysis-{version}.tar.gz"
```
Might be worth adding a small comment about it, so it's clearer what is going on.